californium-core/src/main/java/org/eclipse/californium/core/coap/Option.java

```diff
@@ -319,8 +337,12 @@ public String toString() {
 	}
 	
 	/**
-	 * Renders the option value as string.
-	 * 
+	 * Renders the option value as string. Takes into account of option type,
+	 * thus giving more accurate representation of an option {@code value}.
+	 * Formats {@code value} as integer or string if so defined in
+	 * {@link OptionNumberRegistry}. In case of option {@code value} is just
+	 * an opaque byte array, formats this value as hex string.
+	 *
 	 * @return the option value as string
 	 */
 	public String toValueString() {
```

Comment:
I think `toDescription()` would be a better name. But this may be proposed after that PR.